File: 1018.cs
using System;

class MainClass {
  public static void Main (string[] args) {
    int r = Convert.ToInt32(Console.ReadLine());
    int subs = r;
    int infinity = 1;
    int cen = 0;
    int cin = 0;
    int vin = 0;
    int dez = 0;
    int cinco = 0;
    int dois = 0;
    int um = 0;

    for(int i = 0; i < infinity; i++)
    {
      if(r >= 100)
      {
         cen++;
         r -= 100;
      }
      else if(r >= 50)
      {
        cin++;
        r -= 50;
      }
      else if(r >= 20)
      {
        vin++;
        r -= 20;
      }
      else if(r >= 10)
      {
        dez++;
        r -= 10;
      }
      else if(r >= 5)
      {
        cinco++;
        r -= 5;
      }
      else if(r >= 2)
      {
        dois++;
        r -= 2;
      }
      else if(r >= 1)
      {
        um++;
        r -= 1;
      }
      else
      {
         break;
      }
      infinity++;
    }
    Console.WriteLine(subs);
    Console.WriteLine(cen + " nota(s) de R$ 100,00");
    Console.WriteLine(cin + " nota(s) de R$ 50,00");
    Console.WriteLine(vin + " nota(s) de R$ 20,00");
    Console.WriteLine(dez + " nota(s) de R$ 10,00");
    Console.WriteLine(cinco + " nota(s) de R$ 5,00");
    Console.WriteLine(dois + " nota(s) de R$ 2,00");
    Console.WriteLine(um + " nota(s) de R$ 1,00");
  }
}

File: 1018.cpp
#include <iostream>

using namespace std;

int main() {
  int r[7];
  int t;
  int Count = 100;
  cin >> t;

  for(int i = 0; i < 7; i++)
  {
     r[i] = t / Count;
     t = t % Count;
     if(i == 0)
     Count = 50;
     if(i == 1)
     Count = 20;
     if(i == 2)
     Count = 10;
     if(i == 3)
     Count = 5;
     if(i == 4)
     Count = 2;
     if(i == 5)
     Count = 1;
  }

  cout << t << "\n";
  cout << r[0] << " nota(s) de R$ 100,00\n";
  cout << r[1] << " nota(s) de R$ 50,00\n";
  cout << r[2] << " nota(s) de R$ 20,00\n";
  cout << r[3] << " nota(s) de R$ 10,00\n";
  cout << r[4] << " nota(s) de R$ 5,00\n";
  cout << r[5] << " nota(s) de R$ 2,00\n";
  cout << r[6] << " nota(s) de R$ 1,00\n";
}